pandas/core/arrays/datetimelike.py

```diff
@@ -211,6 +271,10 @@ def astype(self, dtype, copy=True):
     # ------------------------------------------------------------------
     # Null Handling
 
+    def isna(self):
+        # EA Interface
+        return self._isnan
```

Comment:
Same comment as in the other PRs, I think we should keep this concept of `_isnan` on the Index (basically my question about it is: for the array classes, what is the advantages of having a `_isnan` instead of simply using `isna()` everywhere?)  

(but since there are a lot of places here where it is used, I am fine with keeping `_isna` itself in here until we de the actual split)

Reply:
This is a reasonable view, but one that I'd like to put off dealing with for as long as possible.

The eventual move to composition is going to cause a mismatch between what is and is not cacheable (like `_isnan` is now on the Index subclasses).  Since the EA subclasses actually use `_isnan` in arithmetic/comparison ops, there is a performance hit if we're not careful.  Since I haven't thought of a good solution to this, putting it off is the next best thing.